lib/explorer/data_frame.ex

```diff
@@ -998,6 +1012,8 @@ defmodule Explorer.DataFrame do
   Columns are added with keyword list or maps. New variables overwrite existing variables of the
   same name. Column names are coerced from atoms to strings.
 
+  Be aware that mutating grouped dataframes will cause them to lost ordering.
```

Comment:
Small suggestion:

```suggestion
  Be aware that mutating grouped dataframes will cause them to lose ordering.
```

Btw, do you know why this happens? Does it happen with dplyr? In other words, is it a bug or something that will indeed always happen?

Reply:
I'm not sure if happens with `dplyr`, but I will check.

This happens because we have to group the rows, perform the mutation on each group, and then stack the groups together - we lost the order in this step. I don't know if there is a way to keep the original ordering. Probably we could take the indexes of each row, add to a virtual column, sort it and then remove the column :thinking:

Reply:
I see. I think this can actually happen in several cases, not only mutate, so I would rather document that certain operations in groups will change the current ordering.

Reply:
@josevalim I could fix the ordering here: https://github.com/elixir-nx/explorer/pull/238/commits/5ba2dd11feb1eb5af9599c5cbd4b7df6725a18ed

I don't know how we are going to do for other backends, but at least the default one is fixed :)